osbs/core.py

```diff
@@ -786,61 +792,60 @@ def create_image_stream(self, stream_json):
         check_response(response)
         return response
 
-    def import_image(self, name):
+    def update_image_stream(self, stream_id, stream_json):
+        url = self._build_url("imagestreams/%s" % stream_id)
+        response = self._put(url, data=json.dumps(stream_json),
+                             use_json=True)
+        check_response(response)
+        return response
+
+    @retry_on_conflict
+    def import_image(self, name, stream_import):
         """
         Import image tags from a Docker registry into an ImageStream
 
-        :return: bool, whether new tags were imported
+        :return: bool, whether tags were imported
         """
 
         # Get the JSON for the ImageStream
-        url = self._build_url("imagestreams/%s" % name)
-        imagestream_json = self._get(url).json()
+        imagestream_json = self.get_image_stream(name).json()
         logger.debug("imagestream: %r", imagestream_json)
-        spec = imagestream_json.get('spec', {})
-        if 'dockerImageRepository' not in spec:
-            raise OsbsException('No dockerImageRepository for image import')
+
+        if 'dockerImageRepository' in imagestream_json.get('spec', {}):
+            logger.debug("Removing 'dockerImageRepository' from ImageStream %s", name)
+            source_repo = imagestream_json['spec'].pop('dockerImageRepository')
+            imagestream_json['metadata']['annotations'][ANNOTATION_SOURCE_REPO] = source_repo
+            imagestream_json = self.update_image_stream(name, imagestream_json).json()
 
         # Note the tags before import
         oldtags = imagestream_json.get('status', {}).get('tags', [])
         logger.debug("tags before import: %r", oldtags)
 
-        # Mark it as needing import
-        imagestream_json['metadata'].setdefault('annotations', {})
-        check_annotation = "openshift.io/image.dockerRepositoryCheck"
-        imagestream_json['metadata']['annotations'].pop(check_annotation, None)
-        response = self._put(url, data=json.dumps(imagestream_json),
-                             use_json=True)
-        check_response(response)
+        stream_import['metadata']['name'] = name
+        for tag in imagestream_json.get('spec', {}).get('tags', []):
+            image_import = {
+                'from': tag['from'],
+                'to': {'name': tag['name']},
+                'importPolicy': tag.get('importPolicy'),
+                'referenceType': tag.get('referenceType'),
```

Comment:
Should this be `'referencePolicy'`?

Reply:
You're right. I'm wondering where I got `referenceType` from...

Reply:
Fixed.